sympy/matrices/expressions/tests/test_matmul.py

```diff
@@ -153,6 +154,10 @@ def test_construction_with_Mul():
     assert Mul(C, D) == MatMul(C, D)
     assert Mul(D, C) == MatMul(D, C)
 
+def test_construction_with_mul():
+    assert mul(C, D) == MatMul(C, D)
+    assert mul(D, C) == MatMul(D, C)
```

Comment:
I would add `assert mul(C, D) != MatMul(D, C)` as well. It gives better protection against potential (wrong) future edits in `MatMul`.